### PR TITLE
Add LICENCE-STATUS client.

### PR DESCRIPTION
**Pull request by sharpedavid**
Add LICENCE-STATUS to UMS. Allow PIPD to manage LICENCE-STATUS and DMFT-WEBAPP roles.

### CODE REVIEW

File: keycloak-dev/realms/moh_applications/dmft-webapp/main.tf [hunk 1]
```diff
@@ -25,3 +25,25 @@ resource "keycloak_openid_client" "CLIENT" {
...
+
+#module "scope-mappings" {
+#  source    = "../../../../modules/scope-mappings"
+#  realm_id  = keycloak_openid_client.CLIENT.realm_id
+#  client_id = keycloak_openid_client.CLIENT.id
+#  roles = {
+#    "LICENCE-STATUS/MOA"          = var.LICENCE-STATUS.ROLES["MOA"].id,
+#    "LICENCE-STATUS/PRACTITIONER" = var.LICENCE-STATUS.ROLES["PRACTITIONER"].id
+#  }
+#}
```
Comment (sharpedavid) [inline keycloak-dev/realms/moh_applications/dmft-webapp/main.tf:49]: I was getting some errors trying to add the scope mappings. I suspect the roles need to exist before I can use them here. I will finish this PR, and then submit another with this un-commented out.

File: keycloak-dev/realms/moh_applications/pidp-service/main.tf [hunk 2]
```diff
@@ -77,16 +77,17 @@ module "scope-mappings" {
...
-    "USER-MANAGEMENT-SERVICE/view-clients"             = var.USER-MANAGEMENT-SERVICE.ROLES["view-clients"].id,
-    "USER-MANAGEMENT-SERVICE/view-users"               = var.USER-MANAGEMENT-SERVICE.ROLES["view-users"].id,
+    "USER-MANAGEMENT-SERVICE/create-user"                = var.USER-MANAGEMENT-SERVICE.ROLES["create-user"].id,
+    "USER-MANAGEMENT-SERVICE/manage-user-details"        = var.USER-MANAGEMENT-SERVICE.ROLES["manage-user-details"].id,
+    "USER-MANAGEMENT-SERVICE/manage-user-roles"          = var.USER-MANAGEMENT-SERVICE.ROLES["manage-user-roles"].id,
+    "USER-MANAGEMENT-SERVICE/view-client-dmft-webapp"    = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-dmft-webapp"].id,
+    "USER-MANAGEMENT-SERVICE/view-client-licence-status" = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-licence-status"].id,
+    "USER-MANAGEMENT-SERVICE/view-client-hcimweb"        = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-hcimweb"].id,
+    "USER-MANAGEMENT-SERVICE/view-client-pidp-service"   = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-pidp-service"].id,
+    "USER-MANAGEMENT-SERVICE/view-client-sat-eforms"     = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-sat-eforms"].id,
```
Comment (sharpedavid) [inline keycloak-dev/realms/moh_applications/pidp-service/main.tf:87]: The change in code formatting might make this look like I changed the entire block. Use the gear icon at the top to ignore white space changes. I gave the PIPD service account UMS roles on DMFT-WEBAPP and LICENCE-STATUS.

File: keycloak-dev/realms/moh_applications/user-management-service/main.tf [hunk 3]
```diff
@@ -46,8 +46,11 @@ module "client-roles" {
     "view-client-bcer-cp" = {
       "name" = "view-client-bcer-cp"
     },
-    "view-client-dmft-service" = {
-      "name" = "view-client-dmft-service"
+    "view-client-licence-status" = {
```
Comment (ahoplock) [inline keycloak-dev/realms/moh_applications/user-management-service/main.tf:49]: We've been ordering these roles in alphabetical order but I get why it makes sense lumping it together with DMFT.

File: keycloak-dev/realms/moh_applications/pidp-service/main.tf [hunk 4]
```diff
@@ -110,9 +111,13 @@ module "service-account-roles" {
       "client_id" = var.USER-MANAGEMENT-SERVICE.CLIENT.id,
       "role_id"   = "manage-user-roles"
     }
-    "USER-MANAGEMENT-SERVICE/view-client-dmft-service" = {
+    "USER-MANAGEMENT-SERVICE/view-client-dmft-webapp" = {
       "client_id" = var.USER-MANAGEMENT-SERVICE.CLIENT.id,
-      "role_id"   = "view-client-dmft-service"
+      "role_id"   = "view-client-dmft-webapp"
+    }
+    "USER-MANAGEMENT-SERVICE/view-client-licence-status" = {
```
Comment (ahoplock) [inline keycloak-dev/realms/moh_applications/pidp-service/main.tf:118]: See comment below about alphabetical order.

File: keycloak-dev/realms/moh_applications/pidp-service/main.tf [hunk 5]
```diff
@@ -77,16 +77,17 @@ module "scope-mappings" {
...
-    "USER-MANAGEMENT-SERVICE/view-client-pidp-service" = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-pidp-service"].id,
-    "USER-MANAGEMENT-SERVICE/view-client-sat-eforms"   = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-sat-eforms"].id,
-    "USER-MANAGEMENT-SERVICE/view-client-uci-sso"      = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-uci-sso"].id,
-    "USER-MANAGEMENT-SERVICE/view-clients"             = var.USER-MANAGEMENT-SERVICE.ROLES["view-clients"].id,
-    "USER-MANAGEMENT-SERVICE/view-users"               = var.USER-MANAGEMENT-SERVICE.ROLES["view-users"].id,
+    "USER-MANAGEMENT-SERVICE/create-user"                = var.USER-MANAGEMENT-SERVICE.ROLES["create-user"].id,
+    "USER-MANAGEMENT-SERVICE/manage-user-details"        = var.USER-MANAGEMENT-SERVICE.ROLES["manage-user-details"].id,
+    "USER-MANAGEMENT-SERVICE/manage-user-roles"          = var.USER-MANAGEMENT-SERVICE.ROLES["manage-user-roles"].id,
+    "USER-MANAGEMENT-SERVICE/view-client-dmft-webapp"    = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-dmft-webapp"].id,
+    "USER-MANAGEMENT-SERVICE/view-client-licence-status" = var.USER-MANAGEMENT-SERVICE.ROLES["view-client-licence-status"].id,
```
Comment (ahoplock) [inline keycloak-dev/realms/moh_applications/pidp-service/main.tf:84]: See comment below about alphabetical order.

**Review by ahoplock — CHANGES_REQUESTED** [review]
Couple minor things regarding ordering...feel free to disregard if I'm being too nitpick-y here.

**Review by ahoplock — APPROVED** [review]
Reviewed and approved.

**Pull request merged**